File: app/src/main/java/com/BiblioLivro/criarlivros/gestores/GestorVibrator.java
/*
 * Copyright (c) 2020. Está classe está sendo consedida para uso pessoal
 */

package com.BiblioLivro.criarlivros.gestores;

import android.content.Context;
import android.os.Build;
import android.os.VibrationEffect;
import android.os.Vibrator;

import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class GestorVibrator {

    /**
     * @param milliseconds: duração da vibração
     * @param context: receber o SystemService
     */
    public static void Vibrate(long milliseconds, @NotNull Context context)
    {
        Vibrator vibrator = (Vibrator) context.getSystemService(Context.VIBRATOR_SERVICE);

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            Objects.requireNonNull(vibrator).vibrate(VibrationEffect.createOneShot(milliseconds, VibrationEffect.DEFAULT_AMPLITUDE));
        } else
            Objects.requireNonNull(vibrator).vibrate(milliseconds);
    }
}

File: app/src/main/java/com/libry_book/books_catalog/gestores/GestorVibrator.kt
/*
 * Copyright (c) 2023. Está classe está sendo consedida para uso pessoal
 */
package com.libry_book.books_catalog.gestores

import android.content.Context
import android.os.Build
import android.os.VibrationEffect
import android.os.Vibrator
import android.os.VibratorManager

object GestorVibrator {
    /**
     * @param milliseconds: duração da vibração
     * @param context:      receber o SystemService
     */
    @Suppress("DEPRECATION")
    @JvmStatic
    fun vibrate(milliseconds: Long, context: Context) {
        val vibrator: Vibrator
        when {
            Build.VERSION.SDK_INT >= Build.VERSION_CODES.S -> {
                val vibratorManager =
                    context.getSystemService(Context.VIBRATOR_MANAGER_SERVICE) as VibratorManager
                vibrator = vibratorManager.defaultVibrator
                vibrator.vibrate(
                    VibrationEffect.createOneShot(
                        milliseconds,
                        VibrationEffect.DEFAULT_AMPLITUDE
                    )
                )
            }

            Build.VERSION.SDK_INT >= Build.VERSION_CODES.O -> {
                vibrator = context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
                vibrator.vibrate(
                    VibrationEffect.createOneShot(
                        milliseconds,
                        VibrationEffect.DEFAULT_AMPLITUDE
                    )
                )
            }

            else -> {
                vibrator = context.getSystemService(Context.VIBRATOR_SERVICE) as Vibrator
                vibrator.vibrate(milliseconds)
            }
        }
    }
}